pkgs/os-specific/linux/kernel/cachyos-kernels.nix

```diff
@@ -0,0 +1,222 @@
+      PREEMPT = lib.mkOverride 60 yes;
+      PREEMPT_COUNT = yes;
+      PREEMPTION = yes;
+      PREEMPT_DYNAMIC = yes;
+
+      # Enable O3
+      CC_OPTIMIZE_FOR_PERFORMANCE = unset;
+      CC_OPTIMIZE_FOR_PERFORMANCE_O3 = yes;
+
+      # Enable bbr3
+      DEFAULT_CUBIC = unset;
+      TCP_CONG_BBR = yes;
+      DEFAULT_BBR = yes;
+      DEFAULT_TCP_CONG = freeform "bbr";
+
+      # LRU config
+      LRU_GEN = yes;
+      LRU_GEN_ENABLED = yes;
+      LRU_GEN_STATS = unset;
+
+      # VMA config
+      PER_VMA_LOCK = yes;
+      PER_VMA_LOCK_STATS = unset;
+
+      # THP
+      TRANSPARENT_HUGEPAGE_MADVISE = lib.mkOverride 60 unset;
+      TRANSPARENT_HUGEPAGE_ALWAYS = lib.mkOverride 60 yes;
+
+      USER_NS = yes;
+
+      # Modules
+      I2C_NCT6775 = module;
+      STEAMDECK = module;
+    };
+  };
+
+  mainVariant = {
+    version = "6.10.2";
+    variantDescription = "Linux SCHED-EXT + BORE + Cachy Sauce Kernel by CachyOS with other patches and improvements";
+    source = {
+      url = "mirror://kernel/linux/kernel/v6.x/linux-6.10.2.tar.xz";
+      hash = "sha256-c9hSDdnLpaz8XnII52s12XQLiq44IQqSJOMuxMDSm3A=";
+    };
+    patches = [
+      {
+        name = "cachyos-base-all";
+        patch = "${cachyPatches}/6.10/all/0001-cachyos-base-all.patch";
+      }
+      {
+        name = "sched-ext";
+        patch = "${cachyPatches}/6.10/sched/0001-sched-ext.patch";
+      }
+      {
+        name = "bore-cachy-ext";
+        patch = "${cachyPatches}/6.10/sched/0001-bore-cachy-ext.patch";
+      }
+    ];
+
+    config = with lib.kernel; {
+      # CachyOS config
+      CACHY = yes;
+
+      # CPU Scheduler
+      SCHED_CLASS_EXT = yes;
+      SCHED_BORE = yes;
+      MIN_BASE_SLICE_NS = freeform "1000000";
+
+      # LLVM level
+      LTO_NONE = yes;
+
+      # Tick rate
+      HZ_300 = unset;
+      HZ_1000 = yes;
+      HZ = freeform "1000";
+
+      NR_CPUS = lib.mkOverride 60 (freeform "320");
+
+      # Tick type
+      HZ_PERIODIC = unset;
+      NO_HZ_IDLE = unset;
+      CONTEXT_TRACKING_FORCE = unset;
+      NO_HZ_FULL = yes;
+      NO_HZ = yes;
+      NO_HZ_COMMON = yes;
+      CONTEXT_TRACKING = yes;
+
+      # Preempt type
+      PREEMPT_BUILD = yes;
+      PREEMPT_NONE = unset;
+      PREEMPT_VOLUNTARY = lib.mkOverride 60 unset;
+      PREEMPT = lib.mkOverride 60 yes;
+      PREEMPT_COUNT = yes;
+      PREEMPTION = yes;
+      PREEMPT_DYNAMIC = yes;
+
+      # Enable O3
+      CC_OPTIMIZE_FOR_PERFORMANCE = unset;
+      CC_OPTIMIZE_FOR_PERFORMANCE_O3 = yes;
+
+      # Enable bbr3
+      TCP_CONG_CUBIC = lib.mkOverride 60 module;
+      DEFAULT_CUBIC = unset;
+      TCP_CONG_BBR = yes;
+      DEFAULT_BBR = yes;
+      DEFAULT_TCP_CONG = freeform "bbr";
+
+      # Select THP
+      TRANSPARENT_HUGEPAGE_MADVISE = lib.mkOverride 60 unset;
+      TRANSPARENT_HUGEPAGE_ALWAYS = lib.mkOverride 60 yes;
+
+      USER_NS = yes;
+
+      # Modules
+      I2C_NCT6775 = module;
+      NTSYNC = module;
+      V4L2_LOOPBACK = module;
```

Comment:
Per https://github.com/NixOS/nixpkgs/blob/master/pkgs/top-level/linux-kernels.nix#L404 is this compiled anyway?

Reply:
Nixpkgs has a out-of-tree version of v4l2loopback that can be built and installed for the current kernel version by adding it to `boot.extraModulePackages`. That would be roughly equivalent to v4l2loopback-dkms packages on distros like Arch. 

In general, it's better to use in-tree versions of modules whenever they're available and up to date. The CachyOS in-tree version might have some changes that fix build errors or bugs specific to the patched kernel. 

Simply having v4l2loopback in linuxPackages will do no harm, but users of linux-cachyos shouldn't install it. If it builds, it will likely override the module version that comes with the kernel and it might either work worse or the same.

If there is a way to blacklist the extra v4l2loopback package for linux-cachyos or replace it with an empty package, it would probably be the way to go. We'd avoid potential bug reports from users who used the extra module on another kernel and forgot to remove it when switching to linux-cachyos.